### PR TITLE
Allow blacklisting of flash keys which are only used for state

### PR DESCRIPTION
**Pull request by dgmstuart**
This PR adds a config option to specify a blacklist of flash keys to be ignored (fixing #98) 

It also refactors the flash helper to make it more clear and robust

### CODE REVIEW

**Comment by dgmstuart** [comment]
@dsandstrom can you review please and merge if you're happy. 

An alternative approach would be to pass in the ignored flash keys as another parameter to `display_flash_messages`, in the same way as `key_matching`. 

However I feel like the right approach is to prefer to have everything like this in the config (though I'm not sure I have a good argument for why this is good). It would be easy to move the `key_matching` override into the config as a breaking change in a future release.

I considered putting `[:timedout]` as the default value, but I don't think we should have references to devise in the actual code like that.

File: spec/foundation_rails_helper/configuration_spec.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@
 describe FoundationRailsHelper do
 
   describe FoundationRailsHelper::Configuration do
+    # describe "accessors" do
```
Comment (dsandstrom) [inline spec/foundation_rails_helper/configuration_spec.rb]: Remove this comment.

File: README.md [hunk 2]
```diff
@@ -182,6 +182,18 @@ FoundationRailsHelper.configure do |config|
 end
 ```
 
+### Ignored Flash Keys
+The flash helper assumes all flash entries are user-viewable messages.
+To exclude flash entries which are used for storing state
+(e.g. [Devise's `:timedout` flash](https://github.com/plataformatec/devise/issues/1777))
+you can specify a blacklist of keys to ignore with the **ignored_flash_keys** config option:
+```ruby
+FoundationRailsHelper.configure do |config|
```
Comment (dsandstrom) [inline README.md]: Adding another example `FoundationRailsHelper.configure` block seems redundant and possibly confusing.  Maybe have this above the options
```ruby
FoundationRailsHelper.configure do |config|
  # your options here
end
```
then show something like
```ruby
config.ignored_flash_keys
```
below the option's description.

File: lib/foundation_rails_helper/flash_helper.rb [hunk 3]
```diff
@@ -17,12 +17,29 @@ module FlashHelper
...
     def display_flash_messages(key_matching = {})
       key_matching = DEFAULT_KEY_MATCHING.merge(key_matching)
+      key_matching.default = :standard
 
-      flash.inject "" do |message, (key, value)|
-        message += content_tag :div, :data => { :alert => "" }, :class => "alert-box #{key_matching[key.to_sym] || :standard}" do
-          (value + link_to("&times;".html_safe, "#", :class => :close)).html_safe
+      capture do
+        flash.each do |key, value|
+          next if FoundationRailsHelper.configuration.ignored_flash_keys.include? key.to_sym
```
Comment (dsandstrom) [inline lib/foundation_rails_helper/flash_helper.rb]: I prefer this approach over adding a variable to the `display_flash_messages` method because it's more of a global setting and not something that will change in the app.  To be honest, I forgot about the `key_matching` variable.  Maybe we should move that to the config also.

File: spec/foundation_rails_helper/flash_helper_spec.rb [hunk 4]
```diff
@@ -6,32 +6,80 @@
...
-      expect(node).to have_css("div.alert-box.#{foundation_type}", :text => "Flash message")
-      expect(node).to have_css("div.alert-box a.close", :text => "×")
+      expect(node).
+        to  have_css("div.alert-box.#{foundation_type}", :text => "Flash message").
+        and have_css("div.alert-box a.close", :text => "×")
     end
   end
 
+  it "handles symbol keys" do
+    allow(self).to receive(:flash).and_return({ "success" => "Flash message" })
```
Comment (dsandstrom) [inline spec/foundation_rails_helper/flash_helper_spec.rb]: I think you intended:
```
allow(self).to receive(:flash).and_return({ :success => "Flash message" })
```

File: spec/foundation_rails_helper/flash_helper_spec.rb [hunk 5]
```diff
@@ -6,32 +6,80 @@
...
+    it "doesn't raise an error (e.g. NoMethodError)" do
+      allow(self).to receive(:flash).and_return({ "timedout" => true })
+      expect{ Capybara.string display_flash_messages }.not_to raise_error
+    end
+
+    it "doesn't display an alert for that data" do
+      allow(self).to receive(:flash).and_return({ "timedout" => true })
+      expect(display_flash_messages).to be_nil
+
+      # Ideally we'd create a node using Capybara.string, as in the other examples
```
Comment (dsandstrom) [inline spec/foundation_rails_helper/flash_helper_spec.rb]: Expecting nil is fine with me.  Maybe you could have done:
```rspec
expect(node.to_s).to_not have_css("div.alert-box")
```
But, I'm not sure if an empty string responds to css either.

File: lib/foundation_rails_helper/flash_helper.rb (hunk 3 above)
Comment (dgmstuart) [inline lib/foundation_rails_helper/flash_helper.rb]: @dsandstrom a

Comment (dsandstrom) [inline lib/foundation_rails_helper/flash_helper.rb]: Sorry I wasn't clear. Your approach is the way to go.  And I agree that changing the parameter to a config should be a separate PR.  Maybe instead of a breaking change, we allow both ways for one release and deprecate the paramater.  Then remove the parameter on the second release.

Comment (dgmstuart) [inline lib/foundation_rails_helper/flash_helper.rb]: Sounds like a great plan :)

**Comment by dgmstuart** [comment]
@dsandstrom good review :) I've updated based on the comments. I thought it was still worth leaving in the comment in the spec to explain why the style of that case is different from the others.
